docs/design/ovs-pipeline.md

```diff
@@ -1,1197 +1,1938 @@
+them will be redirected to the port `antrea-tc-tap0`, and returned back via the port `antrea-tc-tap1`.
 
-This table is used to determine which "category" of traffic (tunnel, local
-gateway or local Pod) the packet belongs to. This is done by matching on the
-ingress port for the packet. The appropriate value is then written to bits
-[0..3] in NXM_NX_REG0: 0 for tunnel, 1 for local gateway and 2 for local Pod.
-This information is used by matches in subsequent tables. For a packet received
-from the tunnel port, bit 19 in NXM_NX_REG0 is set to 1, to indicate MAC rewrite
-should be performed for the packet in [L3ForwardingTable].
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: TrafficControl
+metadata:
+  name: redirect-web-to-local
+spec:
+  appliedTo:
+    podSelector:
+      matchLabels:
+        app: web
+  direction: Both
+  action: Redirect
+  targetPort:
+    ovsInternal:
+      name: antrea-tc-tap0
+  returnPort:
+    ovsInternal:
+      name: antrea-tc-tap1
+```
 
-If you dump the flows for this table, you may see the following:
+This is a TrafficControl applied to Pods with the label `app: db`. For these Pods, all traffic, both ingress and egress,
+will be mirrored to the port `antrea-tc-tap2`.
 
-```text
-1. table=0, priority=200,in_port=2 actions=set_field:0x1/0xf->reg0,resubmit(,10)
-2. table=0, priority=200,in_port=1 actions=set_field:0/0xf->reg0,load:0x1->NXM_NX_REG0[19],resubmit(,30)
-3. table=0, priority=190,in_port=4 actions=set_field:0x2/0xf->reg0,resubmit(,10)
-4. table=0, priority=190,in_port=3 actions=set_field:0x2/0xf->reg0,resubmit(,10)
-5. table=0, priority=0 actions=drop
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: TrafficControl
+metadata:
+  name: mirror-db-to-local
+spec:
+  appliedTo:
+    podSelector:
+      matchLabels:
+        app: db
+  direction: Both
+  action: Mirror
+  targetPort:
+    ovsInternal:
+      name: antrea-tc-tap2
 ```
 
-Flow 1 is for traffic coming in on the local gateway. Flow 2 is for traffic
-coming in through an overlay tunnel (i.e. from another Node). The next two
-flows (3 and 4) are for local Pods.
+## Egress Implementation
 
-Local traffic then goes to [SpoofGuardTable], while tunnel traffic from other
-Nodes goes to [ConntrackTable]. The table-miss flow entry will drop all
-unmatched packets (in practice this flow entry should almost never be used).
+Table [EgressMark] is dedicated to the implementation of feature `Egress`.
 
-### SpoofGuardTable (10)
+Consider the following Egresses as examples for the remainder of this document.
 
-This table prevents IP and ARP
-[spoofing](https://en.wikipedia.org/wiki/Spoofing_attack) from local Pods. For
-each Pod (as identified by the ingress port), we ensure that:
+This is an Egress applied to Pods with the label `app: web`. For these Pods, all egress traffic will be SNAT'd on Node
+`k8s-node-control-plane` from which we dumped flows in the document with the IP `192.168.77.102`, the transport
```

Comment:
I find it quite confusing that you used the Node's transport IP as the Egress IP, as it is not the typical use case. Maybe we could use a different IP from the same subnet?

Reply:
Update the Egress IP to `192.168.77.112` and `192.168.77.113`. I think this should be a typical use case since the Egress IPs and the Node's transport IPs are in the same CIDR after asking @tnqn for advice.

Reply:
Yes, Egress IPs will typically be from the same subnet as the transport IPs, but will be distinct from transport IPs.